File: moderate/string_rotation.py
#!/usr/bin/env python2
"""
- CodeEval: String Rotation
- https://www.codeeval.com/open_challenges/76/
- This one was annoying as there wasn't any explanation of what "string rotation" was
"""

from sys import argv

def is_rotation(string_1, string_2):
    if len(string_1) != len(string_2):
        return False
    concat_string = string_1 + string_1
    if string_2 in concat_string:
        return True
    else:
        return False


def main():
    with open(argv[1]) as f:
        for line in f:
            string_1, string_2 = line.rstrip('\n').split(',')
            if is_rotation(string_1, string_2):
                print("True")
            else:
                print("False")


if __name__ == "__main__":
    main()

File: moderate/StringRotation.java
/**
 * - CodeEval: String Rotation
 * - https://www.codeeval.com/open_challenges/76/
 */

import java.io.*;

public class StringRotation {

    static boolean isRotation(String string1, String string2)
    {
        if (string1.length() != string2.length())
            return false;
        String concatString = string1 + string1;
        if (concatString.contains(string2))
            return true;
        else
            return false;
    }

    public static void main (String[] args) throws IOException {
        File file = new File(args[0]);
        BufferedReader buffer = new BufferedReader(new FileReader(file));
        String line;
        while ((line = buffer.readLine()) != null) {
            line = line.trim();
            // Process line of input Here
            String[] tokens = line.split(",");
            String string1 = tokens[0];
            String string2 = tokens[1];
            if (isRotation(string1, string2))
                System.out.println("True");
            else
                System.out.println("False");
        }
    }
}
